helper/helper_test.go

```diff
@@ -98,4 +303,169 @@ var _ = Describe("helper tests", func() {
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []SkipFileTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := NewSkipFileTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+		})
+		It("skips batches if skip file is discovered with resize restore", func() {
+			*isResizeRestore = true
+			*origSize = 3
+			*destSize = 5
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []SkipFileTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := NewSkipFileTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+
+		})
+		It("skips batches if skip file is discovered with single datafile", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []SkipFileTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := NewSkipFileTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+		})
```

Comment:
Somehow I don’t see any advantage compared to [those tests](https://github.com/arenadata/gpbackup/pull/103/files#diff-fcd13f1a13de7610af926126f924ee48bfadef4d724ae0186671e3838c291f0eR436-R461) yet.

Reply:
You run only the code you want to test, in the same process. Fast, specific and you may use the debugger.